RFC-lifecycle.md

```diff
@@ -0,0 +1,186 @@
+## The problem
+
+Nock's lifecycle methods are confusingly named and at times inconvenient. It
+takes a lot of studying to figure out the natural thing. It's easy to
+misunderstand this and leave unwanted state in `nock`.
+
+Nock doesn't automatically have a way to assert that mocks have been
+satisfied; it's the caller's responsibility to do this for each one.
+
+See
+https://github.com/paulmelnikow/icedfrisby-nock/blob/master/icedfrisby-nock.js
+for one attempt I made at getting the lifecycle right.
+
+## Typical use cases
+
+1. Assert that all mocks have been satisfied.
+2. Completely reset `nock` after a test.
+3. Allowing unmocked requests only to certain hosts.
+4. Preventing unmocked requests entirely.
+5. Simulating network connection failures.
+6. Temporarily disabling http call interception while preserving registered mocks.
+7. Turn `nock` all the way off and clean up its state (\*\* I've actually never
+   wanted to do this, but wanted to include it in the analysis)
+
+## Analysis
+
+| Use case                                                                | Code                                                                        | Assessment                                                                                                                     |
+| ----------------------------------------------------------------------- | --------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------ |
+| Assert that all mocks have been satisfied                               | `scopes.forEach(scope => scope.done())`                                     | Meh. This requires keeping track of all the scopes.                                                                            |
```

Comment:
Could you maybe remove the `Meh. ` from the assessments? Maybe it’s just me, but somehow it has a negative vibe and I wouldn’t like it to set a tone for communication for the Nock community

Reply:
What I meant was, this interface is passable; it's not confusing but it's not ideal, either. I'll say that instead.

Reply:
On reflection, thanks for the nudge to keep the language sounding positive. 👍